frontend/src/concepts/pipelines/topology/parseUtils.ts

```diff
@@ -159,24 +159,131 @@ export const parseRuntimeInfo = (
+  let completeTime;
+  if (
+    lastUpdatedTime &&
+    (execution.getLastKnownState() === Execution.State.COMPLETE ||
+      execution.getLastKnownState() === Execution.State.FAILED ||
+      execution.getLastKnownState() === Execution.State.CACHED ||
+      execution.getLastKnownState() === Execution.State.CANCELED)
+  ) {
+    completeTime = new Date(lastUpdatedTime).toISOString();
+  }
+  return {
+    startTime: new Date(execution.getCreateTimeSinceEpoch()).toISOString(),
+    completeTime,
+    state: getResourceStateText({
+      resourceType: ResourceType.EXECUTION,
+      resource: execution,
+      typeName: 'Execution',
+    }),
+    taskId: `task.${taskId}`,
+    podName: execution.getCustomPropertiesMap().get('pod_name')?.getStringValue(),
+  };
+};
+
+export enum ResourceType {
+  ARTIFACT = 'ARTIFACT',
+  EXECUTION = 'EXECUTION',
+}
+
+export interface ArtifactProps {
+  resourceType: ResourceType.ARTIFACT;
+  resource: Artifact;
+  typeName: string;
+}
+
+export interface ExecutionProps {
+  resourceType: ResourceType.EXECUTION;
+  resource: Execution;
+  typeName: string;
+}
+
+export type ResourceInfoProps = ArtifactProps | ExecutionProps;
+
+// Get text representation of resource state.
+// Works for both artifact and execution.
+export const getResourceStateText = (props: ResourceInfoProps): string | undefined => {
+  if (props.resourceType === ResourceType.ARTIFACT) {
+    const state = props.resource.getState();
+    switch (state) {
+      case Artifact.State.UNKNOWN:
+        return undefined; // when state is not set, it defaults to UNKNOWN
+      case Artifact.State.PENDING:
+        return 'Pending';
+      case Artifact.State.LIVE:
+        return 'Live';
+      case Artifact.State.MARKED_FOR_DELETION:
+        return 'Marked for deletion';
+      case Artifact.State.DELETED:
+        return 'Deleted';
+      default:
+        return undefined;
+    }
+  } else {
+    // type == EXECUTION
+    const state = props.resource.getLastKnownState();
+    switch (state) {
+      case Execution.State.UNKNOWN:
+        return undefined;
+      case Execution.State.NEW:
+        return 'New';
+      case Execution.State.RUNNING:
+        return 'Running';
+      case Execution.State.COMPLETE:
+        return 'Complete';
+      case Execution.State.CANCELED:
+        return 'Canceled';
+      case Execution.State.FAILED:
+        return 'Failed';
+      case Execution.State.CACHED:
+        return 'Cached';
+      default:
+        return undefined;
+    }
+  }
+};
+
+export const translateStatusForNode = (state?: string): RunStatus | undefined => {
+  switch (state) {
+    case 'Canceled':
     case RuntimeStateKF.CANCELED:
     case RuntimeStateKF.CANCELING:
       return RunStatus.Cancelled;
-    case RuntimeStateKF.PAUSED:
-      return RunStatus.Pending;
+    case 'Running':
+      return RunStatus.Running;
+    case 'Failed':
     case RuntimeStateKF.FAILED:
       return RunStatus.Failed;
+    case 'Pending':
```

Comment:
This is confusing, since the value being returned for the cases you're adding here match. (e.g. `RunStatus.Pending` = "Pending")

I'm not sure it makes sense to use this already unit tested function for this other usage.

Reply:
`RunStatus` is used only for showing the correct icon for topology nodes. It's an enum imported from the PF topology library and be used later. `Pending` here is the status we get from the pipeline run and mlmd context, sometimes the state could be different from the icon we want to show (i.e `RuntimeStateKF.RUNNING` is `RunStatus.InProgress` icon), I think it's good to deal with them separately.